### PR TITLE
JS: newline removal whitelist for js/incomplete-sanitization

### PR DESCRIPTION
**Pull request by esben-semmle**
Addresses the newline part of <https://jira.semmle.com/browse/ODASA-7891>, we now no longer flag:

```js
require("child_process").execSync("which emacs").toString().replace("\n", "")
```

OK'ish [result changes](https://git.semmle.com/esben/dist-compare-reports/tree/js/whitelist-trailing-newline-removal_1554999967413). Eliminates least one [definite FP](https://github.com/mozilla/pdf.js/blob/9128335c4ddf91daabe1b2d4512b9bb207edc652/gulpfile.js#L473), the other eliminated alerts may be TPs if we are unlucky, but I doubt they are security related.

(This will conflict with the upcoming merge-back for #1229)

### CODE REVIEW

**Comment by xiemaisi** [comment]
LGTM, but needs a rebase.

**Comment by esben-semmle** [comment]
rebased

File: javascript/ql/test/query-tests/Security/CWE-116/IncompleteSanitization/tst.js [hunk 1]
```diff
@@ -126,19 +126,12 @@ function good11(s) {
...
+	// motivation for whitelist
+	require("child_process").execSync("which emacs").toString().replace("\n", ""); // OK
 
-	s.replace('[', '\\[').replace(']', '\\]'); // NOT OK
-	s.replace('{', '\\{').replace('}', '\\}'); // NOT OK
-
-	s = s.replace('[', ''); // OK
-	s = s.replace(']', ''); // OK
-	s.replace(/{/, '').replace(/}/, ''); // NOT OK: should have used a string literal if a single replacement was intended
-	s.replace(']', '').replace('[', ''); // probably OK, but still flagged
```
Comment (xiemaisi) [inline javascript/ql/test/query-tests/Security/CWE-116/IncompleteSanitization/tst.js]: This doesn't look right.

Comment (esben-semmle) [reply javascript/ql/test/query-tests/Security/CWE-116/IncompleteSanitization/tst.js]: Fixed.

**Pull request merged**